haystack/components/converters/html.py

```diff
@@ -96,28 +99,56 @@ def run(
         documents = []
         meta_list = normalize_metadata(meta=meta, sources_count=len(sources))
 
-        extractor_class = getattr(extractors, self.extractor_type)
-        extractor = extractor_class(raise_on_failure=False)
+        # Use all extractor types, ensuring user chosen extractor is first, preserve order, avoid duplicates
+        extractors_list = list(
+            dict.fromkeys(
+                [
+                    self.extractor_type,  # User chosen extractor is always tried first
+                    "DefaultExtractor",
+                    "ArticleExtractor",
+                    "ArticleSentencesExtractor",
+                    "LargestContentExtractor",
+                    "CanolaExtractor",
+                    "KeepEverythingExtractor",
+                    "NumWordsRulesExtractor",
```

Comment:
- probably the list of default extractor names can be defined as a constant at the module level instead of being created each time in the `run` method.

- I also think that creating the `extractors_list` (to use) depending on `try_others` would simplify things later: if `try_others is False`, we could just include `self.extractor_type`.